### PR TITLE
HTTP stuff

### PR DESCRIPTION
**Pull request by MattWindsor91**
Basically implements a very rudimentary, GET-only form of the REST API, with some initial websockets stuff.  Hugety huge PRQ by both myself and @wlcx.  Perhaps @LordAro should review :grinning:

### CODE REVIEW

**Comment by wlcx** [comment]
@CaptainHayashi with respect to df9123f - do you think this hilights need for stricter json generation? The full servicestate object was being silently converted to an empty JSON dict

File: connector.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+	reqCh chan httpRequest
+	resCh <-chan baps3.Message
+
+	// TODO(CaptainHayashi): move this away from baps3.Message to
+	// something generic.
+	updateCh chan<- baps3.Message
+}
+
+func initBfConnector(name string, updateCh chan baps3.Message, waitGroup *sync.WaitGroup, logger *log.Logger) (c *bfConnector) {
```
Comment (LordAro) [inline connector.go]: function declaration is a bit long, perhaps shorten some of the variables?

Comment (LordAro) [inline connector.go]: could updateCh get a direction to it? it's only being used in one direction (from the bfConnector), right?

File: connector.go [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+	return
+}
+
+func splitResource(resource string) []string {
+	res := strings.Split(strings.Trim(resource, "/"), "/")
+
+	// The empty resource is returned as {""}: let's fix that
+	if len(res) == 1 && res[0] == "" {
+		res = []string{}
+	}
```
Comment (LordAro) [inline connector.go:96]: this seems very hacky to me..

File: connector.go [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+ * resource to the functions handling them.
+ *
+ * TODO(CaptainHayashi): add support for things that aren't GET
+ *   (have each resource be a jump table of possible methods, or send the method
+ *    to the resMap func?)
+ * TODO(CaptainHayashi): decouple traversal from GET
+ * TODO(CaptainHayashi): maybe make traversal iterative instead of recursive?
+ */
+
+type resMap map[string]func(*bfConnector, []string) interface{}
```
Comment (LordAro) [inline connector.go:129]: ew. I say this, i'm probably going to use something similar for listd at some point

Comment (wlcx) [inline connector.go:129]: You say what?

File: connector.go [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+// controlGet is the GET handler for the /control resource.
+func (c *bfConnector) controlGet(resourcePath []string) interface{} {
+	return c.getResource(controlRes, resourcePath)
+}
+
+// playerGet is the GET handler for the /player resource.
+func (c *bfConnector) playerGet(resourcePath []string) interface{} {
+	// TODO(CaptainHayashi): Probably a spec change, but the fact that this
+	// resource is guarded by more than one feature is iffy.  Do we need a
+	// Player feature?
```
Comment (LordAro) [inline connector.go:184]: i don't think there's an issue with a feature requiring two "FEATURE"s

File: features.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
+package main
+
+// TODO(CaptainHayashi): Petition for move to baps3-go
```
Comment (LordAro) [inline features.go]: pls

**Pull request merged**